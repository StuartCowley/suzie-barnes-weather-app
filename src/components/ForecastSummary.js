import React from "react";
import PropTypes from "prop-types";
import WeatherIcon from "react-icons-weather";

function ForecastSummary({ date, description, icon, temperature, onSelect }) {
  const formattedDate = new Date(date).toDateString();
  return (
    <div className="forecast-summary" data-testid="forecast-summary">
      <div className="forecast-summary__date">{formattedDate}</div>
      <div className="forecast-summary__icon" data-testid="forecast-icon">
        <WeatherIcon name="owm" iconId={icon} />
      </div>
      <div className="forecast-summary__temperature">
        {temperature.max}
        &deg;C
      </div>
      <div className="forecast-summary__description">{description}</div>
      <button type="button" onClick={() => onSelect(date)}>
        More Details...
      </button>
    </div>
  );
}

ForecastSummary.propTypes = {
  date: PropTypes.number.isRequired,
  description: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired,
  onSelect: PropTypes.func.isRequired,
  temperature: PropTypes.shape({
    min: PropTypes.number,
    max: PropTypes.number,
  }).isRequired,
};

export default ForecastSummary;
