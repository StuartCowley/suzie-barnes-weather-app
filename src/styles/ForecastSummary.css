body {
    font-size: 18px;
    text-align: center;
}

.forecast-summary {
    background-color: rgb(255, 226, 64);
    width: auto;
    height: auto;
    border-radius: 10px;
}

.forecast-summary__date {
    font-size: 20px;
    font-weight: bold;
}

button {
    border: none;
}

button:hover {
    background: rgb(173, 172, 172);
    font-weight: bold;
}