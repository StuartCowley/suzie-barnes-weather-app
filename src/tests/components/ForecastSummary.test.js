import React from "react";
import { render } from "@testing-library/react";
import ForecastSummary from "../../components/ForecastSummary";

describe("ForecastSummary", () => {
  const validProps = {
    date: "1525046400000",
    description: "Stub description",
    icon: "stubIcon",
    temperature: {
      min: 11,
      max: 22,
    },
  };
  it("renders correctly", () => {
    const { asFragment } = render(
      <ForecastSummary
        date={validProps.date}
        description={validProps.description}
        icon={validProps.icon}
        temperature={validProps.temperature}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});

// expect(getByText("1525046400000")).toBeInstanceOf(className="forecast-summary__date")
// expect(getByText("Clear")).toBeInstanceOf(className="forecast-summary__description")
// expect(getByText("11")).toBeInstanceOf(className="forecast-summary__temperature")
// expect(getByTestid(className="forecast-summary__icon"))
