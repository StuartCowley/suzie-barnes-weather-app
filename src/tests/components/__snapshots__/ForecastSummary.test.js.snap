// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummary renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-summary"
  >
    <div
      class="forecast-summary__date"
    >
      1525046400000
    </div>
    <div
      class="forecast-summary__icon"
    >
      stubIcon
    </div>
    <div
      class="forecast-summary__temperature"
    >
      22Â°C
    </div>
    <div
      class="forecast-summary__description"
    >
      Stub description
    </div>
  </div>
</DocumentFragment>
`;
